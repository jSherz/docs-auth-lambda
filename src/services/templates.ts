import { promisify } from "util";
import { GOOGLE_CLIENT_ID, REDIRECT_URL } from "../vars";

import * as fs from "fs";

const fsReadFile = promisify(fs.readFile);

export const renderTemplate = async (error: string = "") => {
  const template = await fsReadFile("public/login.html");

  const url =
    "https://accounts.google.com/o/oauth2/v2/auth?" +
    "scope=email&" +
    "include_granted_scopes=true&" +
    `redirect_uri=${REDIRECT_URL}&` +
    "response_type=code&" +
    `client_id=${GOOGLE_CLIENT_ID}`;

  return template
    .toString("utf8")
    .replace("__LOGIN_URL__", url)
    .replace("__ERROR__", error)
    .replace(
      "__ERROR_STYLE__",
      error && error.length >= 1 ? "" : "display: none",
    );
};
