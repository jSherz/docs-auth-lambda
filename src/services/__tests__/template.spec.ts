import { renderTemplate } from "../templates";

describe("services/template", () => {
  describe("renderTemplate", () => {
    it("renders a template without an error", async () => {
      await expect(renderTemplate()).resolves.toMatchSnapshot();
    });

    it("renders a template with an error", async () => {
      await expect(
        renderTemplate("bad thingz happened"),
      ).resolves.toMatchSnapshot();
    });
  });
});
