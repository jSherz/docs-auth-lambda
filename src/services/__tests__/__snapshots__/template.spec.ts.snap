// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`services/template renderTemplate renders a template with an error 1`] = `
"<!doctype html>
<html lang=\\"en\\">
<head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">

    <title>Login</title>

    <link rel=\\"stylesheet\\" href=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css\\"
          integrity=\\"sha384-9gVQ4dYFwwWSjIDZnLEWnxCjeSWFphJiwGPXr1jddIhOegiu1FwO5qRGvFXOdJZ4\\" crossorigin=\\"anonymous\\">

    <style type=\\"text/css\\">
        .container {
            margin-top: 2rem;
        }
    </style>
</head>

<body>
<main role=\\"main\\">
    <section class=\\"text-center\\">
        <div class=\\"container\\">
            <div class=\\"row justify-content-center\\">
                <div id=\\"login\\" class=\\"col-lg-6 align-self-center\\">
                    <div class=\\"card\\">
                        <div class=\\"card-body\\">
                            <h1 class=\\"card-title\\">Authentication Required</h1>
                            <p class=\\"card-text text-muted\\">
                                Sign in with your GSuite account.
                            </p>
                            <p>
                                <a href=\\"https://accounts.google.com/o/oauth2/v2/auth?scope=email&include_granted_scopes=true&redirect_uri=https://d2isfcdxdi4wm9.cloudfront.net/login.php3&response_type=code&client_id=674222348431-8kgaftfbumtqcr9b5cpn7880104u5n0h.apps.googleusercontent.com\\" class=\\"btn btn-primary\\">Login</a>
                            </p>
                            <div class=\\"alert alert-danger\\" role=\\"alert\\" style=\\"\\">bad thingz happened</div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </section>
</main>

<script src=\\"https://code.jquery.com/jquery-3.3.1.slim.min.js\\"
        integrity=\\"sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo\\"
        crossorigin=\\"anonymous\\"></script>
<script src=\\"https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.0/umd/popper.min.js\\"
        integrity=\\"sha384-cs/chFZiN24E4KMATLdqdvsezGxaGsi4hLGOzlXwp5UZB1LY//20VyM2taTB4QvJ\\"
        crossorigin=\\"anonymous\\"></script>
<script src=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/js/bootstrap.min.js\\"
        integrity=\\"sha384-uefMccjFJAIv6A+rW+L4AHf99KvxDjWSu1z9VI8SKNVmz4sk7buKt/6v9KI65qnm\\"
        crossorigin=\\"anonymous\\"></script>
</body>
</html>
"
`;

exports[`services/template renderTemplate renders a template without an error 1`] = `
"<!doctype html>
<html lang=\\"en\\">
<head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">

    <title>Login</title>

    <link rel=\\"stylesheet\\" href=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css\\"
          integrity=\\"sha384-9gVQ4dYFwwWSjIDZnLEWnxCjeSWFphJiwGPXr1jddIhOegiu1FwO5qRGvFXOdJZ4\\" crossorigin=\\"anonymous\\">

    <style type=\\"text/css\\">
        .container {
            margin-top: 2rem;
        }
    </style>
</head>

<body>
<main role=\\"main\\">
    <section class=\\"text-center\\">
        <div class=\\"container\\">
            <div class=\\"row justify-content-center\\">
                <div id=\\"login\\" class=\\"col-lg-6 align-self-center\\">
                    <div class=\\"card\\">
                        <div class=\\"card-body\\">
                            <h1 class=\\"card-title\\">Authentication Required</h1>
                            <p class=\\"card-text text-muted\\">
                                Sign in with your GSuite account.
                            </p>
                            <p>
                                <a href=\\"https://accounts.google.com/o/oauth2/v2/auth?scope=email&include_granted_scopes=true&redirect_uri=https://d2isfcdxdi4wm9.cloudfront.net/login.php3&response_type=code&client_id=674222348431-8kgaftfbumtqcr9b5cpn7880104u5n0h.apps.googleusercontent.com\\" class=\\"btn btn-primary\\">Login</a>
                            </p>
                            <div class=\\"alert alert-danger\\" role=\\"alert\\" style=\\"display: none\\"></div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </section>
</main>

<script src=\\"https://code.jquery.com/jquery-3.3.1.slim.min.js\\"
        integrity=\\"sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo\\"
        crossorigin=\\"anonymous\\"></script>
<script src=\\"https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.0/umd/popper.min.js\\"
        integrity=\\"sha384-cs/chFZiN24E4KMATLdqdvsezGxaGsi4hLGOzlXwp5UZB1LY//20VyM2taTB4QvJ\\"
        crossorigin=\\"anonymous\\"></script>
<script src=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/js/bootstrap.min.js\\"
        integrity=\\"sha384-uefMccjFJAIv6A+rW+L4AHf99KvxDjWSu1z9VI8SKNVmz4sk7buKt/6v9KI65qnm\\"
        crossorigin=\\"anonymous\\"></script>
</body>
</html>
"
`;
